backend/authentication/auth_controller.js

```diff
@@ -150,6 +150,143 @@ exports.githubCallback = async (req, res) => {
   }
 };
 
+//Update User Profile
+exports.updateProfile = async (req, res) => {
+  const { username } = req.params;
+  const { newFullName, newImageUrl, availability, skills, categories } =
+    req.body;
+
+  try {
+    const user = await prisma.user.findUnique({
+      where: {
+        username: username,
+      },
+    });
+    if (!user) {
+      return res.status(404).json({
+        error: "User not found",
+      });
+    }
+
+    let newProfileData = {};
+
+    if (newFullName) {
+      newProfileData.fullName = newFullName;
+    }
+    if (newImageUrl) {
+      newProfileData.imageUrl = newImageUrl;
+    }
+    if (availability) {
+      newProfileData.availability = availability;
+    }
+
+    const updatedUserProfile = await prisma.user.update({
+      where: {
+        username: username,
+      },
+      data: newProfileData,
+    });
+
+    if (categories && categories.length > 0) {
+      await prisma.user.update({
+        where: {
+          username: username,
+        },
+        data: {
+          categories: {
+            set: [],
+          },
+        },
+      });
+      for (category of categories) {
+        try {
+          await prisma.category.create({
+            data: {
+              category: category,
+              User: {
+                connect: {
+                  username: username,
+                },
+              },
+            },
+          });
+        } catch (error) {
+          res.status(500).json({
+            error: "Error updating categories",
+          });
+        }
+      }
+    }
+    if (skills && skills.length > 0) {
+      await prisma.user.update({
+        where: {
+          username: username,
+        },
+        data: {
+          skills: {
+            set: [],
+          },
+        },
+      });
```

Comment:
Also, I see we are calling the prisma.user.update method three times within this function. It might be possible to consolidate these calls into a single update. However, I understand this might be more complex than using `createMany`, so please feel free to ignore this comment if you think it's too difficult.

Reply:
Since this is essentially deleting the values of the skills data, I used deleteMany to batch delete and createMany to batch create. I assumed that the consolidation might make it a bit less readable. Should I still do it? Is there an efficiency tradeoff with the new method in https://github.com/izuchukwumba/peered/pull/8/commits/5fe0755ce9404ede218e0bb4bcfe4fbc765b2e99 ?